### ISSUE TITLE
Spurious linefeeds in rich messages

### ISSUE THREAD

**Issue by ara4n** [vector-im/element-x-android#1078]
### Steps to reproduce

1. Send a rich text message (via markdown)
2. Note that it's displayed with unexpected linefeeds after the style changes.

e.g.

<img width="453" alt="Screenshot 2023-08-17 at 22 43 50" src="https://github.com/vector-im/element-x-android/assets/1294269/0c3d68af-2117-49e1-bb1f-b15372586d47">

but should be:

<img width="453" alt="Screenshot 2023-08-17 at 22 43 50" src="https://github.com/vector-im/element-x-android/assets/1294269/d87c209f-fe19-48b8-91c3-339c5b7a7dbc">

```json
    "body": "> <@piroko:pintobyte.com> and the occasional history gap bug\n\nhm, there *definitely* shouldn't be history gaps - please report the hell out of them if you see any",
    "format": "org.matrix.custom.html",
    "formatted_body": "<mx-reply><blockquote><a href=\"https://matrix.to/#/!aCvOWKNDeXXpPgquKq:matrix.org/$ltT1taNn77J4EypG25t2fH45cLWnaYdDBtkgv2JBkR4?via=matrix.org&via=envs.net&via=element.io\">In reply to</a> <a href=\"https://matrix.to/#/@piroko:pintobyte.com\">@piroko:pintobyte.com</a><br>and the occasional history gap bug</blockquote></mx-reply>hm, there <em>definitely</em> shouldn't be history gaps - please report the hell out of them if you see any",
```

#### What did you expect?

no spurious linefeeds in rich text msgs

### Your phone model

S10

### Operating system version

12

### Application version and app store

0.1.2 (40001020) playstore

### Will you send logs?

No

### Are you willing to provide a PR?

No

**Comment by anoadragon453** [comment]
This still occurs on 4749197d046ddc9dbcbe2270ee580e648c7ac112:

![image](https://github.com/vector-im/element-x-android/assets/1342360/2129e88d-e6b1-48d5-acce-829243f0651e)

![image](https://github.com/vector-im/element-x-android/assets/1342360/b3fa8a35-6ddd-4899-a43c-b568d9d2efda)

Content of the `m.room.message` event:

```
"content": {
    "body": "Is there a database or something like that where `nixpkgs` packages can be looked up by their contents, like it is possible at https://www.debian.org/distrib/packages ?",
    "format": "org.matrix.custom.html",
    "formatted_body": "Is there a database or something like that where <code>nixpkgs</code> packages can be looked up by their contents, like it is possible at https://www.debian.org/distrib/packages ?",
    "m.mentions": {},
    "msgtype": "m.text"
  }
```

**Comment by bxdxnn** [comment]
I can't reproduce this with both messages, so it either depends on DPI or was already fixed.

**Comment by frebib** [comment]
I can reliably reproduce this with strike through words. I'm on a Pixel 8 Pro and I have `smallest width` set to 460dp in developer options ![Screenshot_20260408-215237.png](https://github.com/user-attachments/assets/6c842313-6800-41e8-b0eb-8ebd055a98eb)

**Comment by bxdxnn** [comment]
> I can reliably reproduce this with strike through words. I'm on a Pixel 8 Pro and I have `smallest width` set to 460dp in developer options ![Screenshot_20260408-215237.png](https://github.com/user-attachments/assets/6c842313-6800-41e8-b0eb-8ebd055a98eb)

I can reproduce it with this one.
It also works in the RTE, there's a linefeed spawning when you add a strikethrough, and the cursor misaligns.

**Comment by jmartinesp** [comment]
This is a bug caused by a Jsoup upgrade that considers `del`, `code` and other inline tags block tags instead for some reason, and the rendering code in EXA has some logic like: `if (element.isBlock() and !isLineBreak(element)) text.append('\n')` which adds line breaks after *actual* block tags like paragraphs, quotes or code blocks, but thanks to this weird behaviour change, does so for some inline tags too.

**Comment by bmarty** [comment]
Maybe jsoup 1.22.1 fixes it, but we refuse the upgrade for another reason (see https://github.com/element-hq/element-x-android/pull/5965). Maybe worth trying to upgrade the lib first?

**Comment by jmartinesp** [comment]
No, I don't think it does. I checked [the source code](https://github.com/jhy/jsoup/blob/d4a8941820c037327538c30a8723ec715b67b6f6/src/main/java/org/jsoup/parser/TagSet.java#L241) and `del` is still considered a block tag 🫤 . I'm trying writing our own method to check this: https://github.com/element-hq/matrix-rich-text-editor/pull/118
